lib/SIL/Verifier/SILVerifier.cpp

```diff
@@ -5403,6 +5417,55 @@ void SILVTable::verify(const SILModule &M) const {
               "vtable entry for " + baseName + " must be ABI-compatible",
               *entry.Implementation);
     }
+    
+    // Validate the entry against its superclass vtable.
+    if (!superclass) {
+      // Root methods should not have inherited or overridden entries.
+      bool validKind;
+      switch (entry.TheKind) {
+      case Entry::Normal:
+      case Entry::NormalNonOverridden:
+        validKind = true;
+        break;
+        
+      case Entry::Inherited:
+      case Entry::Override:
+        validKind = false;
+        break;
+      }
+      assert(validKind && "vtable entry in root class must not be inherited or override");
```

Comment:
continue and eliminate the else? I.e.:

```
if (...) {
   assert(validKind, ...)
   continue;
}

if (superVTable) {
   ...
}
```

Reply:
The reason why I am suggesting this is that it makes it clearer that the if statement will never cause any code beneath it to be evaluated. Without this, I have to scan after the else if to see if that code may be run after the first if completes. In contrast, with the early continue like this I now have a nice invariant I can use when scanning the code below. Take it or leave it ; ).